modules/apps/object/object-service/src/main/java/com/liferay/object/internal/item/selector/SystemObjectEntryItemSelectorView.java

```diff
@@ -225,10 +252,18 @@ public String getTitle(Locale locale) {
 				return StringPool.BLANK;
 			}
 
-			Map<String, Object> modelAttributes =
-				_baseModel.getModelAttributes();
+			User user = _userLocalService.fetchUser(
+				PrincipalThreadLocal.getUserId());
+
+			Map<String, Object> values = ObjectEntryDTOConverterUtil.toDTO(
+				_baseModel, _dtoConverterRegistry, _objectDefinition.getName(),
+				_systemObjectDefinitionManagerRegistry, user);
 
-			return (String)modelAttributes.get(objectField.getDBColumnName());
+			return String.valueOf(
+				ObjectEntryValuesUtil.getTitleFieldValue(
+					objectField.getBusinessType(),
+					_baseModel.getModelAttributes(), objectField.getName(),
+					user, values));
```

Comment:
@seltong before we were doing `modelAttributes.get(objectField.getDBColumnName())`, but you changed to `modelAttributes.get(objectField.getName())` [here](https://github.com/liferay-objects/liferay-portal/blob/133d0457a11475c773ebadb443df8b2270c048e8/modules/apps/object/object-api/src/main/java/com/liferay/object/entry/util/ObjectEntryValuesUtil.java#L46). 

Make sure we're not changing the behavior that was already working.